compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -366,8 +366,12 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                                 }),
                             }
                         }
-                        thir::InlineAsmOperand::Const { expr } => mir::InlineAsmOperand::Const {
-                            value: unpack!(block = this.as_local_operand(block, expr)),
+                        thir::InlineAsmOperand::Const { value } => mir::InlineAsmOperand::Const {
+                            value: box Constant {
+                                span: expr_span,
```

Comment:
Doesn't this cover the whole `asm!()`?

Reply:
I suppose I could add a `span` field to `thir::InlineAsmOperand::Const` to pass the span through...